src/VisualStudio/CSharp/Impl/Options/Formatting/FormattingOptionPageControl.xaml

```diff
@@ -25,6 +25,27 @@
                     <CheckBox x:Name="FormatOnPasteCheckBox" x:Uid="FormatOnPasteCheckBox" />
                 </StackPanel>
             </GroupBox>
+            <GroupBox x:Name="FormatDocumentSettingsGroupBox" x:Uid="FormatDocumentSettingsGroupBox" >
+                <StackPanel x:Uid="StackPanel_2">
+                    <CheckBox x:Name="IsCodeCleanupConfiguredCheckBox" x:Uid="IsCodeCleanupConfiguredCheckBox" Visibility="Hidden" Height="0"/>
```

Comment:
why is this hidden?

Reply:
It tracks the state of whether the user visited this configuration page and clicked OK since the feature was implemented. It's one of two ways we know not to show the gold bar (the other being Don't Show This Again).

Reply:
Sorry, i'm not getting it.  Why do we need "hidden UI" for this purpose?  If we want to track the state of what the user did/didn't do on the page, why don't we use any of our existing tracking mechanism... is there some reason they're not sufficient?